src/integration_test/java/com/klodnicki/Bike/v2/rest/controller/AdminChargingStationControllerIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+
+import com.klodnicki.Bike.v2.DTO.station.ListStationsForAdminResponseDTO;
+import com.klodnicki.Bike.v2.DTO.station.StationForAdminResponseDTO;
+import com.klodnicki.Bike.v2.model.BikeType;
+import com.klodnicki.Bike.v2.model.entity.Bike;
+import com.klodnicki.Bike.v2.model.entity.ChargingStation;
+import com.klodnicki.Bike.v2.repository.BikeRepository;
+import com.klodnicki.Bike.v2.repository.ChargingStationRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(locations = "/application-test.properties")
+class AdminChargingStationControllerIntegrationTest {
+
+    @Autowired
+    private WebTestClient webTestClient;
+    @Autowired
+    private ChargingStationRepository chargingStationRepository;
+    @Autowired
+    private BikeRepository bikeRepository;
+    @Autowired
+    private ModelMapper modelMapper;
+    private ChargingStation chargingStation;
+    private Bike bike;
+
+    @BeforeEach
+    public void setUp() {
+        chargingStation = new ChargingStation(null, "station name", "station address", "station city",
+                100, new ArrayList<>());
+        chargingStationRepository.save(chargingStation);
+
+        bike = new Bike(null, BikeType.ELECTRIC, null, null, null);
+        bikeRepository.save(bike);
+    }
+
+    @Test
+    public void addBikeToList_ShouldAddBikeToStationListOfBikesAndReturnChargingStation_WhenStationIdAndBikeIdAreGiven() {
+        List<Bike> bikeList = new ArrayList<>();
+        bikeList.add(bike);
+        chargingStation.setBikeList(bikeList);
+
+        webTestClient.post()
+                .uri(uriBuilder -> uriBuilder
+                        .path("/api/admin/stations/list/add/" + chargingStation.getId())
+                        .queryParam("bikeId", bike.getId())
+                        .build())
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(ChargingStation.class)
+                .consumeWith(response -> {
+                            ChargingStation station = response.getResponseBody();
+                            assertNotNull(station);
+                            assertFalse(station.getBikeList().isEmpty());
+                            assertIterableEquals(chargingStation.getBikeList(), station.getBikeList());
+                        }
+                );
+    }
+
+    @Test
+    public void add_ShouldAddChargingStationToDatabaseAndReturnStationForAdminResponseDTO_WhenChargingStationIsGiven() {
```

Comment:
I think this test is missing checking if the charging station is saved in the database while the name of the test suggest it should do it.